core/org.eclipse.cdt.core.tests/misc/org/eclipse/cdt/internal/core/envvar/CMakeBuildEnvironmentSupplierTest.java

```diff
@@ -0,0 +1,254 @@
+		IEnvironmentVariable var;
+		ExtendedCMakeBuildConfiguration cmBuildConfig = new ExtendedCMakeBuildConfiguration(buildConfig,
+				"cmBuildConfigName", mockToolchain, toolChainFile);
+		ICoreEnvironmentVariableSupplier sup = EnvironmentVariableManager.fCmakeSupplier;
+		// context is null and name not "PATH"
+		var = sup.getVariable(pathVariableName, null);
+		assertNull(var);
+		var = sup.getVariable("notPATH", null);
+		assertNull(var);
+		// context is CMakeBuildConfiguration
+		var = sup.getVariable(pathVariableName, cmBuildConfig);
+		assertEquals(EnvironmentVariableManager.getDefault().getDefaultDelimiter(), var.getDelimiter());
+		assertEquals(pathVariableName, var.getName());
+		assertEquals(IEnvironmentVariable.ENVVAR_PREPEND, var.getOperation());
+		assertTrue(var.getValue().contains(cmakeLocation));
+		assertTrue(var.getValue().contains(generatorLocation[0]));
+
+	}
+
+	@Test
+	public void test_3_CMakeBuildEnvironmentSupplier_getVariables() {
+		IEnvironmentVariable[] vars;
+		ExtendedCMakeBuildConfiguration cmBuildConfig = new ExtendedCMakeBuildConfiguration(buildConfig,
+				"cmBuildConfigName", mockToolchain, toolChainFile);
+		ICoreEnvironmentVariableSupplier sup = EnvironmentVariableManager.fCmakeSupplier;
+		// context is null
+		vars = sup.getVariables(null);
+		assertArrayEquals(vars, new IEnvironmentVariable[0]);
+		assertThat(Arrays.asList(vars), hasSize(0));
+		// context is CMakeBuildConfiguration
+		vars = sup.getVariables(cmBuildConfig);
+		assertThat(Arrays.asList(vars), hasSize(1));
+		assertEquals(EnvironmentVariableManager.getDefault().getDefaultDelimiter(), vars[0].getDelimiter());
+		assertEquals("PATH", vars[0].getName());
+		assertEquals(IEnvironmentVariable.ENVVAR_PREPEND, vars[0].getOperation());
+		assertTrue(vars[0].getValue().contains(cmakeLocation));
+		assertTrue(vars[0].getValue().contains(generatorLocation[0]));
+	}
+
+	@Test
+	public void test_4_CmakePathAddedInOrder() {
+		ExtendedCMakeBuildConfiguration cmBuildConfig = new ExtendedCMakeBuildConfiguration(buildConfig,
+				"cmBuildConfigName", mockToolchain, toolChainFile);
+
+		Properties environmentVariables = EnvironmentReader.getEnvVars();
+		Map<String, String> env = new HashMap<>();
+		for (String key : environmentVariables.stringPropertyNames()) {
+			String value = environmentVariables.getProperty(key);
+			env.put(key, value);
+		}
+		cmBuildConfig.setBuildEnvironment(env);
+		// CMake tool locations must be added in order: ISV > CMake > System's path
+		String pathStr = env.get(pathVariableName);
+		assertNotNull(pathStr);
+		String regex = String.join(EnvironmentVariableManager.getDefault().getDefaultDelimiter(), isvPath,
+				cmakeLocation, generatorLocation[0], ".*").replace("\\", "\\\\"); //$NON-NLS-1$
+		Pattern pattern = Pattern.compile(regex);
+		assertTrue("CMake tools' location not found between ISV variable's path and System variable's path",
+				pattern.matcher(pathStr).find());
```

Comment:
I find this code a little hard to read. A simpler approach than making a regex to try to match is to split on the path separator and check the indexes are as expected:

```suggestion
		var list = Arrays.asList(pathStr.split(File.pathSeparator)).subList(0, 3);
		assertThat(list, is(List.of(isvPath, cmakeLocation, generatorLocation[0])));
```

Alternatively, you can just do a startsWith:

```java
assertThat(pathStr, startsWith(String.join(File.pathSeparator, isvPath, cmakeLocation, generatorLocation[0])));
```